engine/resources/file.go

```diff
@@ -137,9 +137,19 @@ func (obj *FileRes) isDir() bool {
 // the case where the mode is not specified. The caller should check obj.Mode is
 // not empty.
 func (obj *FileRes) mode() (os.FileMode, error) {
-	m, err := strconv.ParseInt(obj.Mode, 8, 32)
+	n, err := strconv.ParseInt(obj.Mode, 8, 32)
+	m := os.FileMode(n)
 	if err != nil {
-		return os.FileMode(0), errwrap.Wrapf(err, "mode should be an octal number (%s)", obj.Mode)
+		// Try parsing symbolic
+		modes := strings.Split(obj.Mode, ",")
+		m = os.FileMode(0)
```

Comment:
If this is the initial condition, don't we want this to be the mode of the current file since we might +/- on it?

Reply:
If there is an error getting the mode from the file should it be raised then or just set to 0 and have that handled elsewhere?

Reply:
The first one. So if when trying to read the file mode that errors, then we propagate that error up and the engine handles it.